### DISCUSSION TITLE
TextStyle

### DISCUSSION THREAD

**Discussion by dickermoshe** [nank1ro/flutter-shadcn-ui#4] [General]
What are your plans for TextStyle?

**Comment by nank1ro** [comment]
We've two options:
- Create a custom TextTheme class with `h1`, `h2`, `h3`, `h4`, `p`, `blockquote`, `table` and so on
- Map the text styles given by Material to the shadcn text styles, for example `bodyMedium` -> `p` and set the default values in the `Theme` created by `ShadcnApp`

**Reply by dickermoshe** [reply]
Seems like a good poll for Twitter.  

Creating a custom `TextTheme` object would give us the opportunity to fix whatever pain points that developers have with the materials.
The `GoogleFonts` package is really popular, it would be a shame if it couldn't be used.

**Comment by dickermoshe** [comment]
Implementing a `ColorTheme` and `TextTheme` before a large subsection of the widgets are created would be foolhardy.
It's impossible to know at such an early stage what's needed. 
I'll gladly help with the refactoring when you're ready for that.

**Reply by nank1ro** [reply]
What do you mean with `ColorTheme`?
Mike said that will help to provide the direction on how to map the ShadcnThemeData to the material ThemeData, even for the component themes.

You can start creating the `ShadcnTextTheme` and provide also a method like `toMaterial()` to convert it into a `TextTheme`.
Thanks for your help 🙏

**Comment by dickermoshe** [comment]
@nank1ro What do you think about using a platform agnostic font?
https://vercel.com/font

**Reply by nank1ro** [reply]
@dickermoshe why should we select a font in a UI library, where is the end user to choose the preferred font family?

**Reply by dickermoshe** [reply]
The default font changes based on the platform running the app.
This is one of the pain points material has . It pretends to be adaptive but it's really not.
So you randomly find things that don't look the same because Flutter was "nice" to you.

**Reply by dickermoshe** [reply]
This may be a bad solution. But I think this pain point could use a cure.

**Reply by nank1ro** [reply]
Never noticed these font changes in my apps, probably because I always used a custom font family.
Feel free to open a PR, I may add the `fontFamily` parameter to the ThemeData and pass it to the default textTheme.
The important step is that the font family should be customizable.
What's your handle on Twitter? I could mention you when I post something you've contributed

**Reply by dickermoshe** [reply]
@dickerdev

**Comment by nank1ro** [comment]
Implemented the Geist Sans font family as default font family, like shadcn. The font family can still be changed by the end user if needed.
